geode-core/src/main/java/org/apache/geode/internal/cache/BaseRegionMap.java

```diff
@@ -0,0 +1,156 @@
+import org.apache.geode.internal.logging.LogService;
+
+/**
+ * BaseRegionMap consolidates common behaviour between {@link AbstractRegionMap} and
+ * {@link ProxyRegionMap}. {@link AbstractRegionMap} should be extended when implementing
+ * a new {@link RegionMap} and {@link ProxyRegionMap} should be used if an empty {@link RegionMap}
+ * for regions whose DataPolicy is Proxy is required.
+ */
+abstract class BaseRegionMap implements RegionMap {
+  private static final Logger logger = LogService.getLogger();
+
+  @Override
+  public void incRecentlyUsed() {
+    // nothing
+  }
+
+  @Override
+  public long getEvictions() {
+    return 0;
+  }
+
+  @Override
+  public EvictionController getEvictionController() {
+    return null;
+  }
+
+  @Override
+  public void lruUpdateCallback() {
+    // nothing needed
+  }
+
+  @Override
+  public boolean lruLimitExceeded(DiskRegionView diskRegionView) {
+    return false;
+  }
+
+  @Override
+  public void lruCloseStats() {
+    // nothing needed
+  }
+
+  @Override
+  public void resetThreadLocals() {
+    // By default do nothing; LRU maps needs to override this method
+  }
+
+
+  @Override
+  public boolean disableLruUpdateCallback() {
+    // nothing needed
+    return false;
+  }
+
+  @Override
+  public void enableLruUpdateCallback() {
+    // By default do nothing; LRU maps needs to override this method
+  }
+
+  @Override
+  public int centralizedLruUpdateCallback() {
+    return 0;
+  }
+
+  @Override
+  public void updateEvictionCounter() {}
+
+  @Override
+  public void finishChangeValueForm() {}
+
+  @Override
+  public boolean beginChangeValueForm(EvictableEntry le,
+      CachedDeserializable vmCachedDeserializable, Object v) {
+    return false;
+  }
+
+  @Override
+  public void lruEntryFaultIn(EvictableEntry entry) {
+    // do nothing by default
+  }
+
+  /**
+   * If true then invalidates that throw EntryNotFoundException or that are already invalid will
+   * first call afterInvalidate on CacheListeners. The old value on the event passed to
+   * afterInvalidate will be null. If the region is not initialized then callbacks will not be done.
+   * This property only applies to non-transactional invalidates. Transactional invalidates ignore
+   * this property. Note that empty "proxy" regions on a client will not be sent invalidates from
+   * the server unless they also set the proxy InterestPolicy to ALL. If the invalidate is not sent
+   * then this property will not cause a listener on that client to be notified of the invalidate. A
+   * non-empty "caching-proxy" will receive invalidates from the server.
+   */
+  @MutableForTesting
+  public static boolean FORCE_INVALIDATE_EVENT =
+      Boolean.getBoolean(DistributionConfig.GEMFIRE_PREFIX + "FORCE_INVALIDATE_EVENT");
+
+  /**
+   * If the FORCE_INVALIDATE_EVENT flag is true then invoke callbacks on the given event.
+   */
+  static void forceInvalidateEvent(EntryEventImpl event, LocalRegion owner) {
```

Comment:
If you change these package-private static methods to non-static, the subclasses can invoke them without the BaseRegionMap. prefix. I recommend changing all of these to non-static:
```
void forceInvalidateEvent(EntryEventImpl event, LocalRegion owner)

boolean shouldInvokeCallbacks(final LocalRegion owner, final boolean isInitialized)

EntryEventImpl switchEventOwnerAndOriginRemote(EntryEventImpl event, boolean originRemote)
```

Reply:
The `BaseRegionMap` prefix seems to just be an artifact of the refactoring. It can be safely removed even for static methods as they are available in both subclasses. 
It might be clearer to keep them static as it shows that the functionality has nothing to do with an actual RegionMap instance. (They are functional helper functions for the `RegionMap`)
Will cleanup the  `BaseRegionMap` prefixes